src/main/kotlin/org/rust/ide/intentions/MatchToIfLetIntention.kt

```diff
@@ -57,3 +62,14 @@ class MatchToIfLetIntention : RsElementBaseIntentionAction<MatchToIfLetIntention
         get() = (this is RsBlockExpr && block.lbrace.getNextNonCommentSibling() == block.rbrace)
             || this is RsUnitExpr
 }
+
+private fun containsStdOptionOrResult(item: RsEnumItem?, pattern: RsPat): Boolean {
+    if (item?.isStdOptionOrResult != true) return false
+    return with(pattern.skipUnnecessaryTupDown().text) {
+        if (item == item.knownItems.Option) {
```

Comment:
This code seems similar to `isExhaustiveOptionOrResult`.

Reply:
Made a separate function.